packages/indexer/src/plugins/kv.ts

```diff
@@ -1,128 +1,137 @@
 import assert from "node:assert";
 import type { Cursor, DataFinality } from "@apibara/protocol";
-import { type Database, type ISqlite, open } from "sqlite";
+import Database, { type Database as SqliteDatabase } from "better-sqlite3";
 import { useIndexerContext } from "../context";
 import { deserialize, serialize } from "../vcr";
 import { defineIndexerPlugin } from "./config";
 
-type SqliteArgs = ISqlite.Config;
+type SqliteArgs = Database.Options & {
+  filename: string | Buffer | undefined;
+};
 
 export function kv<TFilter, TBlock, TRet>(args: SqliteArgs) {
```

Comment:
What if the user simply passes the `Database` directly? That way they can configure wal if they want or use different (but API compatible) implementations.

Reply:
Well, surely we can have the user pass Database directly.
Also `better-sqlite3 `has slightly different API implementation than what we had earlier in `sqlite` & `sqlite3` package. So people must have to use `better-sqlite3`.